src/services/drpcEvents/index.ts

```diff
@@ -146,4 +161,90 @@ export default class DrpcEvents {
       })
     }
   }
+
+  private async handleSubmitQueryResponse(request: DrpcRequest, agentConnectionId: string) {
+    let queryId: string | null = null
+    try {
+      this.logger.info(
+        'DRPC response (%s) received on connection %s of method %s',
+        request.id,
+        agentConnectionId,
+        request.method
+      )
+
+      let params: SubmitQueryResponseRPCParams
+      try {
+        params = submitQueryResponseRpcParams.parse(request.params)
+      } catch (err) {
+        this.logger.warn('Invalid parameters received for request %s: %o', request.id, request.params)
+        await this.cloudagent.submitDrpcResponse(request.id, {
+          error: {
+            code: drpcErrorCode.INVALID_PARAMS,
+            message: `invalid params object`,
+          },
+        })
+        return
+      }
+      const [connection] = await this.db.get('connection', { agent_connection_id: agentConnectionId })
+      if (!connection) {
+        this.logger.warn('Invalid connection for drpc message %s', agentConnectionId)
+        return
+      }
+      //find corresponding query based on queryIdForResponse provided by responder,
+      //this should match a query id in our db
+      const [queryRow] = await this.db.get('query', { id: params.queryIdForResponse })
+      if (!queryRow) {
+        this.logger.warn('Invalid queryId for drpc message %s', params.queryIdForResponse)
+        return
+      }
+      if (queryRow.query_response !== null) {
+        this.logger.warn('It appears this query: %s has already been answered.', params.queryIdForResponse)
+        return
+      }
+
+      //update corresponding query
+      this.logger.warn(queryRow)
+      const [query] = await this.db.update(
+        'query',
+        { id: queryRow.id },
+        { query_response: params.emissions, status: 'resolved' }
+      ) //do we want to save the full params object?
+
+      queryId = query.id
+      const result = {
+        state: 'accepted',
+      }
+      await this.cloudagent.submitDrpcResponse(request.id, { result })
+      await this.db.insert('query_rpc', {
+        query_id: query.id,
+        method: 'submit_query_response',
+        role: 'server',
+        agent_rpc_id: request.id,
+        result,
+      })
+    } catch (err) {
+      if (err instanceof Error) {
+        this.logger.warn('Error thrown whilst processing DRPC request %s', err.message)
+      } else {
+        this.logger.warn('Unknown error thrown whilst processing DRPC request')
+        this.logger.trace(`err: %o`, err)
+      }
+
+      if (queryId !== null) {
+        await this.db.update(
+          'query',
+          { id: queryId },
+          {
+            status: 'errored',
+          }
+        )
+      }
```

Comment:
I've thought about this and we should remove this update. We don't want to invalidate a query because of an unknown error. I think the lesser evil here is to leave the ability for the other side to provide an answer

Reply:
Agreed, removed and left a comment in the code in regards to fire and forget.